include/qpmad/solver.h

```diff
@@ -894,8 +894,15 @@ namespace qpmad
         }
 
 
-        template <int t_rows, int t_cols, int... t_Other>
-        static void factorizeCholeskyInPlace(Eigen::Matrix<t_Scalar, t_rows, t_cols, t_Other...> &H)
+        template <int... t_Other>
+        static void factorizeCholeskyInPlace(Eigen::Matrix<t_Scalar, t_Other...> &H)
+        {
+            const Eigen::LLT<Eigen::Ref<typename std::decay<decltype(H)>::type>, Eigen::Lower> llt(H);
+            QPMAD_UTILS_PERSISTENT_ASSERT(
+                    Eigen::Success == llt.info(), "Could not perform Cholesky decomposition of the Hessian (dense).");
+        }
+
+        static void factorizeCholeskyInPlace(Eigen::Ref<Eigen::MatrixXd> &H)
```

Comment:
Maybe you don't need `&`.

https://stackoverflow.com/questions/21132538/correct-usage-of-the-eigenref-class/21138316#21138316
> In general, using a non const reference like Ref<T>& is never a good idea because this will only work if the caller already has a Ref<T> object at hand. This discards 5 and 7.

Reply:
In this case it is not going to work anyway since `solve()` overloads take a reference -> https://github.com/asherikov/qpmad/blob/master/include/qpmad/solver.h#L175. When I have time I'll try to revise API to handle Eigen::Ref properly.